From a3b0290e8c0f30e2bece4bb76cbf80a84ee235b7 Mon Sep 17 00:00:00 2001
From: Stefan Roese <sr@denx.de>
Date: Thu, 23 May 2019 13:39:16 +0200
Subject: [PATCH] MIPS: ralink: dts: gardena_smart_gateway_mt7688: Add ethernet
 and ESW DT nodes

We can not upstream this because there is no DSA switch driver yet in
mainline.

Signed-off-by: Stefan Roese <sr@denx.de>
Upstream-Status: Inappropriate [for OpenWRT driver]
(cherry picked from commit ca752fde19951c02f2a96c196a9006346446e3d2)
---
 .../boot/dts/ralink/gardena_smart_gateway_mt7688.dts     | 9 +++++++++
 arch/mips/boot/dts/ralink/mt7628a.dtsi                   | 2 ++
 2 files changed, 11 insertions(+)

diff --git a/arch/mips/boot/dts/ralink/gardena_smart_gateway_mt7688.dts b/arch/mips/boot/dts/ralink/gardena_smart_gateway_mt7688.dts
index 0bfb1dde9764..7eb927a30722 100644
--- a/arch/mips/boot/dts/ralink/gardena_smart_gateway_mt7688.dts
+++ b/arch/mips/boot/dts/ralink/gardena_smart_gateway_mt7688.dts
@@ -116,6 +116,7 @@ led-ethernet-activity {
 
 	aliases {
 		serial0 = &uart0;
+		ethernet0 = &ethernet;
 	};
 };
 
@@ -171,6 +172,14 @@ nand_flash@1 {
 	};
 };
 
+&ethernet {
+	status = "okay";
+};
+
+&esw {
+	status = "okay";
+};
+
 &uart1 {
 	status = "okay";
 
diff --git a/arch/mips/boot/dts/ralink/mt7628a.dtsi b/arch/mips/boot/dts/ralink/mt7628a.dtsi
index 589baeb3b289..5b90d5cf09ac 100644
--- a/arch/mips/boot/dts/ralink/mt7628a.dtsi
+++ b/arch/mips/boot/dts/ralink/mt7628a.dtsi
@@ -315,6 +315,8 @@ ethernet: ethernet@10100000 {
 		reset-names = "fe", "esw";
 
 		status = "disabled";
+
+		mediatek,switch = <&esw>;
 	};
 
 	esw: esw@10110000 {
-- 
2.43.0

